src/app-init/router.js

```diff
@@ -81,79 +81,79 @@ routerConfig(
-      { name: ROUTE_PROFILE_TYPE_LIST, path: '/profiletype' },
-      { name: ROUTE_PROFILE_TYPE_ADD, path: '/profiletype/add' },
-      { name: ROUTE_PROFILE_TYPE_EDIT, path: '/profiletype/edit/:profiletypeCode' },
-      { name: ROUTE_PROFILE_TYPE_ATTRIBUTE_ADD, path: '/profiletype/attribute/:entityCode/add' },
-      { name: ROUTE_PROFILE_TYPE_ATTRIBUTE_EDIT, path: '/profiletype/attribute/:entityCode/edit/:attributeCode' },
-      { name: ROUTE_ATTRIBUTE_MONOLIST_PROFILE_ADD, path: '/profiletype/attribute/:entityCode/MonolistAdd/:attributeCode' },
-      { name: ROUTE_USER_LIST, path: '/user' },
-      { name: ROUTE_USER_ADD, path: '/user/add' },
-      { name: ROUTE_USER_EDIT, path: '/user/edit/:username' },
-      { name: ROUTE_USER_DETAIL, path: '/user/view/:username' },
-      { name: ROUTE_USER_RESTRICTIONS, path: '/user/restrictions' },
-      { name: ROUTE_USER_MY_PROFILE, path: '/myProfile' },
-      { name: ROUTE_GROUP_LIST, path: '/group' },
-      { name: ROUTE_GROUP_ADD, path: '/group/add' },
-      { name: ROUTE_GROUP_DETAIL, path: '/group/view/:groupname' },
-      { name: ROUTE_GROUP_EDIT, path: '/group/edit/:groupCode' },
-      { name: ROUTE_USER_AUTHORITY, path: '/authority/:username' },
-      { name: ROUTE_LABELS_ADD, path: '/labels/add' },
-      { name: ROUTE_LABELS_AND_LANGUAGES, path: '/labels-languages' },
-      { name: ROUTE_LABEL_ADD, path: '/labels-languages/add' },
-      { name: ROUTE_LABEL_EDIT, path: '/labels-languages/edit/:labelCode' },
-      { name: ROUTE_CATEGORY_LIST, path: '/category' },
-      { name: ROUTE_CATEGORY_ADD, path: '/category/add' },
-      { name: ROUTE_CATEGORY_EDIT, path: '/category/edit/:categoryCode' },
-      { name: ROUTE_CATEGORY_DETAIL, path: '/category/detail/:categoryCode' },
-      { name: ROUTE_ROLE_LIST, path: '/role' },
-      { name: ROUTE_ROLE_ADD, path: '/role/add/' },
-      { name: ROUTE_ROLE_EDIT, path: '/role/edit/:roleCode' },
-      { name: ROUTE_ROLE_DETAIL, path: '/role/view/:roleCode' },
-      { name: ROUTE_RELOAD_CONFIG, path: '/reloadConfiguration' },
-      { name: ROUTE_RELOAD_CONFIRM, path: '/reloadConfiguration/confirm' },
-      { name: ROUTE_PLUGIN_CONFIG_PAGE, path: '/plugin/:pluginId/config' },
-      { name: ROUTE_DATABASE_LIST, path: '/database' },
-      { name: ROUTE_DATABASE_ADD, path: '/database/add' },
-      { name: ROUTE_DATABASE_REPORT, path: '/database/report/:dumpCode' },
-      { name: ROUTE_DATABASE_DUMP_TABLE, path: '/database/report/dumpTable/:dumpCode' },
-      { name: ROUTE_FILE_BROWSER, path: '/file-browser' },
-      { name: ROUTE_FILE_BROWSER_UPLOAD, path: '/file-browser/upload' },
-      { name: ROUTE_FILE_BROWSER_CREATE_FOLDER, path: '/file-browser/create-folder' },
-      { name: ROUTE_FILE_BROWSER_CREATE_TEXT_FILE, path: '/file-browser/create-text-file' },
+      { name: ROUTE_HOME, path: `${process.env.PUBLIC_URL}/` },
```

Comment:
I think this might be less invasive and error prone, if we just added a new config option to this router config object, e.g.:

```pathPrefix: process.env.PUBLIC_URL,```

and modified the router to add it to the paths of the compiled routes RegExps

Reply:
I should have left more context to what's going on here:

The goal of these changes is to allow us to package an "all-in-one" distribution of Entando that has a pre-built App Builder bundled with the Java application.  To do this, we need to add a fix to the App Builder to enable it to run in sub-folder and to use a relative path for the DOMAIN.  I wanted to make this not break the current way of doing things either.

1. To run in a sub-folder, I am adding the PUBLIC_URL to the route table.  PUBLIC_URL is a special env variable that will automatically be set up.  We only need to provide a value if we want it to run in a subfolder (example below).

2. To allow a relative path to the Mapp engine, I introduced a new env variable called EMBEDDED_DOMAIN, which can refer to the context of the Mapp engine.  If this is set, it will dynamically append the value of this variable to the base url of the application.  If not, the DOMAIN variable will work as before.

Here are the env files for example:

Example 1 ("as before"):
USE_MOCKS=false
DOMAIN=http://localhost:8081/entando

Example 2 (Embedded example):
USE_MOCKS=false
PUBLIC_URL=/appbuilder
EMBEDDED_DOMAIN=entando

Example 2 would result in the following working (wherever it is deployed):
http://server.name:port/appbuilder
http://server.name:port/entando

Hope this clears things up

Reply:
Here is a sample application that uses this method when using mvn jetty:run (and also includes a custom landing page at the root url): https://github.com/kspokas/entando-pam-sample/tree/appbuilder-experiment

Reply:
Makes sense to me. I'd like @nicpuddu to review as well before merging

Reply:
I do agree with @alessioscalici on this, I'd rather not have the process variable used over and over, it is more error prone.

in theory it would be better adding a prefix parmeter to the router component in general, so that we don't have to deal with this in each route, and set it when configuring the router.

Reply:
Makes sense, I will refactor this and update the readme